test/unit/server_test.js

```diff
@@ -275,19 +275,20 @@ describe("server.js tests", function () {
         it("throws a NotFoundError", function (done) {
           this.axiosMock.expects('get')
             .withArgs(sinon.match('https://horizon-live.stellar.org:1337/ledgers/1'))
-            .returns(Promise.reject({status: 404, data: {}}));
+            .returns(Promise.reject({response: {status: 404, data: {}}}));
 
           this.server.ledgers()
             .ledger(1)
             .call()
             .then(function () {
               done("didn't throw an error");
             })
-            .catch(StellarSdk.NotFoundError, function (err) {
-              done();
-            })
             .catch(function (err) {
-              done(err);
+              if (err instanceof StellarSdk.NotFoundError) {
```

Comment:
`bluebird` Promise API is incompatible with `Promise` spec. You can't do something like using native `Promise`:
```js
            .catch(StellarSdk.NotFoundError, function (err) {
              done();
            })
```
We're using `babel-plugin-transform-builtin-extend` not to allow users to check if an error is an instance of some error, like: `if (err instanceof StellarSdk.NotFoundError)`.